src/gt4py/next/program_processors/runners/dace/transformations/map_fusion_extended.py

```diff
@@ -459,6 +460,32 @@ def can_be_applied(
 
         # TODO(phimuell): Implement a check that the array is single use data?
 
+        # Check that the data read by the second map from the intermediate AccessNode
+        # are not copied to another AccessNode.
+        access_node_out_edges = list(graph.out_edges(self.access_node))
+        access_node_out_edges_data = [
+            oedge.src.data
+            for oedge in access_node_out_edges
+            if isinstance(oedge.dst, dace_nodes.AccessNode)
+        ]
+        second_map_in_edges = list(graph.in_edges(self.second_map_entry))
+        matching_data_subsets = {
+            second_map_edge.src.data: second_map_edge.data.src_subset
+            for second_map_edge in second_map_in_edges
+            if second_map_edge.src.data in access_node_out_edges_data
+            and isinstance(second_map_edge.src, dace_nodes.AccessNode)
```

Comment:
```suggestion
            if isinstance(second_map_edge.src, dace_nodes.AccessNode)
            and second_map_edge.src.data in access_node_out_edges_data
```
Technically you also have to check if the `dst_conn` starts with `IN_` because it could also be [Dynamic Map Ranges](https://spcldace.readthedocs.io/en/latest/sdfg/ir.html#parametric-parallelism).
I think it is safe to ignore them, if want to be on the safe side, then check if you have any of them and if so, do not apply.